### PR TITLE
(0.95.3) Restores open boundary condition functionality

### PR DESCRIPTION
**Pull request by jagoosw**
The refactoring in https://github.com/CliMA/Oceananigans.jl/pull/3834 broke flat extrapolation open boundary conditions. This restores them.

Closes https://github.com/CliMA/Oceananigans.jl/issues/3975

### CODE REVIEW

**Comment by glwagner** [comment]
shoudl we add a test to make sure it keeps working?

**Comment by simone-silvestri** [comment]
a test is a good idea.

**Comment by jagoosw** [comment]
I might be missing something but I think that the changes in https://github.com/CliMA/Oceananigans.jl/pull/3834 make it so that open halos are filled even when `fill_boundary_normal_velocities` is set to `false`. I can't see what mechanism would stop that happening @simone-silvestri ?

File: src/BoundaryConditions/fill_halo_regions.jl [hunk 1]
```diff
@@ -20,12 +20,14 @@ conditions, possibly recursing into `fields` if it is a nested tuple-of-tuples.
...
 
+retrieve_bc(bc) = bc
+
 # Returns the boundary conditions a specific side for `FieldBoundaryConditions` inputs and
 # a tuple of boundary conditions for `NTuple{N, <:FieldBoundaryConditions}` inputs
 for dir in (:west, :east, :south, :north, :bottom, :top)
     extract_side_bc = Symbol(:extract_, dir, :_bc)
     @eval begin
-        @inline $extract_side_bc(bc) = bc.$dir
+        @inline $extract_side_bc(bc) = retrieve_bc(bc.$dir)
```
Comment (jagoosw) [inline src/BoundaryConditions/fill_halo_regions.jl:30]: @simone-silvestri I think we need this analogously to `retrieve_open_bc`

**Comment by jagoosw** [comment]
@glwagner @simone-silvestri I think this should be all done now and we should merge ASAP since open boundaries are currently broken

**Review by simone-silvestri — APPROVED** [review]
sounds good!

**Comment by jagoosw** [comment]
Please could someone rerun these tests now Sverdrup is working again

**Comment by tomchor** [comment]
Let's maybe try to release both this and https://github.com/CliMA/Oceananigans.jl/pull/3606 together to avoid multiple patch releases. Both seem to be ready to merge.

**Comment by jagoosw** [comment]
Happy for you to merge

**Comment by tomchor** [comment]
> Happy for you to merge

Sounds good! I'll merge this and I should merge https://github.com/CliMA/Oceananigans.jl/pull/3606 today too. Then I'll tag a new version.

**Pull request merged**